src/store/cartSlice.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice, current } from '@reduxjs/toolkit'
+import { RootState } from '.'
+import Product from '../interfaces/Product'
+
+export interface CartProduct {
+  id?: string;
+  sku?: string;
+  name?: string;
+  mainimage: {
+    alt: string;
+    copyright?: string;
+    dimensions: {
+      height: number;
+      weight: number;
+    };
+    url: string;
+  };
+  price: number;
+  stock: number;
+  ammount: number;
+}
+
+const initialState = {
+  products: [] as Array<CartProduct>
+}
+
+export const cartSlice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    setCart: (state, { payload }) => {
+      const alreadyInCart = state.products.find((product) => product?.id === payload.id)
```

Comment:
For this case is more convenient the `.findIndex` function, right now you are doing `.find` and then `.indexOf`